Change how arguments passed are counted

jeparlefrancais opened the pull request:
Closes #103 

I added a .gitattributes so that on checkout, git does not change line endings for the `.stderr` files. Otherwise it makes all the lint tests fail if someone has `core.autocrlf` on. I'm not that familiar with that kind of stuff but from the test I made I think it works.

This is an attempt at fixing this use case:
```lua
assert(t.string(variable))
```
The standard library enforces that `assert` is called with two arguments, which I think is a good idea. But it does not work well when using a library like [`t`](https://github.com/Roblox/t), that uses function that returns the two values.

#### What I did
The way I solved it is by checking if the last value passed to the function is a `...` or a function call. Then if it is, it updates the minimum amount of values passed to the function. So it is still possible to know if there are too much arguments.

#### Another option
Another way I thought about doing it was to change the `get_argument_type` function to return `ArgumentType::Vararg` in the function call case. After that it's possible to just check if the last argument type is `Vararg` to determine if the max number of arguments is unknown or not.

I don't know if that could have a larger impact on other stuff so that's why I went with what I did.

#### Bad side of the fix
The sad thing is that it allows weird things to pass. Like this for example:
```
print( type(print(), print()) )
```
Which is technically correct, but horrible.

Let me know your thoughts!

Kampfkarren commented:
> The sad thing is that it allows weird things to pass. Like this for example:
> 
> print( type(print(), print()) )

What causes this? This seems like something that should be simple to block.

Kampfkarren commented on selene-lib/tests/lints/standard_library/assert.lua:
```diff
@@ -0,0 +1,4 @@
+assert(true, 'message')
```
Can you replace this and the other one with double quotes just to be consistent?

jeparlefrancais commented:
> What causes this? This seems like something that should be simple to block.

Yes, you are right! I changed it so it assumes that a function call or `...` give at least one argument. I just added a flag to know if there are potentially more arguments.

Kampfkarren commented:
Can you add that to the tests now that it works?

Kampfkarren commented:
Thanks a bunch!
